canvas/templatetags/canvas.py

```diff
@@ -15,4 +15,29 @@ def is_allowed_to_open_course(course, user):
 
 @register.simple_tag
 def is_allowed_to_open_event(event, user):
-    return event.has_view_permission(user)
+    return event.has_view_permission(user) or event.status == "Closed"
```

Comment:
I think we have a is_allowed_to_open function in the event model class. this logic should not be here. You can modify it or create that function if we don't have it and call that function here.